### PR TITLE
Parse complex optargs with sequential types

### LINKED ISSUE #544: Pod `*err*` doesn't get redirected to cider
I noticed that calls to `boot.util/info` from inside a pod do not print in a cider repl.  Further investigation revealed that if you `boot repl` from a command line and `cider-connect` to it, then pod output goes to the command line, and non-pod output goes to cider.  I'm hoping there's some way to redirect pod output so that it goes to the same place.

### PR DESCRIPTION
**Pull request by arichiardi**
This patch enables to parse options using complex optargs like `SYM:VER=PATH`
with types that are not nested but straight up sequential, like `[sym str str]`.

### CODE REVIEW

File: boot/core/build.boot [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
 
 (import boot.App)
 
-(ns-unmap 'boot.user 'test)
-
-(deftask test []
-  (boot.util/info "Testing against version %s\n" (App/config "BOOT_VERSION"))
-  (comp (runtests)
+(deftask integration-test []
+  (set-env! :source-paths #{"integration-test"})
```
Comment (arichiardi) [inline boot/core/build.boot]: I am always tripping on this one. It is not working, will fix it.

**Comment by arichiardi** [comment]
Don't understand why the tests are failing in this very weird way:

```shell
FAIL in boot.cli-test/cli-tests (cli_test.clj:43)                               
parse-cli
[fix #578] split arguments with two different separators using vector optarg.
expected: {:parent [prj "1.3" "parent"]}
  actual: {:parent [ prj "1.3" "parent"]}
    diff: - {:parent [prj]}
          + {:parent [ prj]}
```

File: CHANGES.md [hunk 2]
```diff
@@ -34,6 +34,7 @@ and `technomancy` for the above explanation.
 - Bump [pomegranate](https://github.com/cemerick/pomegranate) and [dynapath](https://github.com/tobias/dynapath) to `1.0.0`. [#612][612]
 - Digest `java.io.File` instead `String` path of jar at sift-action `:add-jar` method [#678][678]
 - Add warning about improper use of repl :eval option [#666][666]
+- Parse complex optargs with sequential types [682][682]
```
Comment (martinklepsch) [inline CHANGES.md]: Maybe add an example here what was previously possible and whats possible now.

File: boot/core/src/boot/cli.clj [hunk 3]
```diff
@@ -43,7 +43,17 @@
...
+  "Return a function accepting arguments that are split according to optarg.
+
+  The function works also for optargs splitting:
+
+    ((parse-fn SYM:VER=PATH) SYM:VER=PATH) => [SYM VER PATH]
+
+  But the typical usage is:
+
+    ((parse-fn SYM:VER=PATH) \"prj:1.3=parent\") => [prj 1.3 parent]"
+  [optarg]
```
Comment (martinklepsch) [inline boot/core/src/boot/cli.clj]: Would this also work?
```
((parse-fn SYM:VER=PATH=) \"prj:1.3=parent=xxx\")
```

Comment (arichiardi) [reply boot/core/src/boot/cli.clj]: Probably no because the parsing needs `A-Z` interleaved by symbols at the moment

File: CHANGES.md (hunk 2 above)
Comment (arichiardi) [reply CHANGES.md]: Ok I will add one

File: boot/core/src/boot/cli.clj (hunk 3 above)
Comment (arichiardi) [reply boot/core/src/boot/cli.clj]: Yes it works, I tried it by having a test, one second I am going to push it

File: CHANGES.md [hunk 4]
```diff
@@ -34,6 +34,7 @@ and `technomancy` for the above explanation.
 - Bump [pomegranate](https://github.com/cemerick/pomegranate) and [dynapath](https://github.com/tobias/dynapath) to `1.0.0`. [#612][612]
 - Digest `java.io.File` instead `String` path of jar at sift-action `:add-jar` method [#678][678]
 - Add warning about improper use of repl :eval option [#666][666]
+- Parse complex optargs with sequential types, see [wiki section][sequential-optargs] [#682][682]
```
Comment (martinklepsch) [inline CHANGES.md]: That's perfect 👌

**Comment by martinklepsch** [comment]
Merged https://github.com/boot-clj/boot/commit/197f991324d9207f5d4b7b1828dd54ff61aadd09 https://github.com/boot-clj/boot/commit/69e2957c05023b312296f4f8940ed88b78ebaa59 https://github.com/boot-clj/boot/commit/9b21dd98d18c96d856250c0b7555b8a660f7704e

**Comment by martinklepsch** [comment]
I need to revisit my workflow for merging PRs bit annoyed that this is picked up as closed instead of merged now. 😏

**Pull request closed without merging**